File: algo/intersection-of-two-arrays-ii/intersect.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        hm, ret = {}, []
        for n in nums1:
            if n not in hm:
                hm[n] = 1
            else:
                hm[n] += 1
        for n in nums2:
            if n in hm and hm[n] > 0:
                hm[n] -= 1
                ret.append(n)
        return ret

File: algo/intersection-of-two-arrays-ii/intersect.go
func intersect(nums1 []int, nums2 []int) []int {
    hm := map[int]int{}
    ret := []int{}
    for _, n := range nums1 {
        if _, ok := hm[n]; ok {
            hm[n] += 1
        } else {
            hm[n] = 1
        }
    }
    for _, n := range nums2 {
        if times, ok := hm[n]; ok && times > 0 {
            hm[n] -= 1
            ret = append(ret, n)
        }
    }
    return ret
}